_posts/en/2018-04-23-practical-applications-of-webhooks.md

```diff
@@ -0,0 +1,135 @@
+
+- **action** – a certain event/behavior that is to be executed,
+- **trigger** – defines when an action is to be executed,
+- **callback** – determines in what way the client application is to be informed about the action’s results.
+
+A full documentation of available actions, triggers and callbacks is located in [this GitHub repository](https://github.com/Consdata/sonarqube-companion/wiki/Webhooks)
+
+As an example, let's define a webhook that will check how the number of violations within a team has changed over the last day. Depending on the result, an appropriate message will be sent to the messenger’s channel.
+
+First, define what action should result in sending the message. To do this, define an action for a new webhook in the webhooks node:
+```json
+"webhooks": [{
+    "action": {
+    "type": "NO_IMPROVEMENT",
+    "period": "DAILY",
+    "severity": ["blockers", "criticals", "majors"]
+    }
+}]
+```
+In this way, you have defined an action that will check whether the number of violations in the group's projects has improved during the last day. Additionally, it will take into account only the violations with the following priorities: blocker, critical and major.
+
+The next step is to define an action trigger. In our example, for simplicity, we would like the action to be executed every minute. To achieve that, define a CRON-type trigger:
+```json
+"trigger": {
+    "type": "CRON",
+    "definition": "0 */1 * * * *"
+},
+```
+
+Finally, you need to define the actual integration with the messenger. To do that, define a POST-type callback, including the URL obtained from the "Webhook URL" field of Slack’s configuration panel.
+```json
+"callbacks": [
+    {
+        "type": "POST",
+        "url": "https://hooks.slack.com/services/*/*",
+        "body" : {
+            "no_improvement": "{ 'text': 'http://gph.is/1RFg2r3 Brak poprawy'}",
+            "improvement": "{ 'text': 'http://gph.is/1a7RlDR Poprawiono ${diff}'",
+            "clean": "{ 'text': 'https://gph.is/1IH3RW6 Czysto'}"
+        }
+    }
+]
+```
+
+In the body section you need to define the content of the message to be sent depending on the state of the group. For example, if the state of violations has improved (the "improvement" response), a gif will be sent along with a short comment containing a predefined action variable - ${diff}. In this way, the channel will show information about the number of corrected violations during the last day.
+
+The whole configuration looks as follows:
+```json
+"webhooks": [
+    {
+        "action": {
+        "type": "NO_IMPROVEMENT",
+        "period": "DAILY",
+        "severity": ["blockers", "criticals", "majors"]
+        },
+        "trigger": {
+            "type": "CRON",
+            "definition": "0 */1 * * * *"
+        },
+        {
```

Comment:
Z tym JSONem jest coś nie teges chyba

Reply:
to powinnio być wewnątrz "callbacks" : [ ]

Reply:
To poprawię w obu wersjach językowych posta